inc/misc.h

```diff
@@ -0,0 +1,4 @@
+#pragma once
+// Common helper macros
+
+#define SIZEOF_ARRAY(arr) (sizeof((arr))/sizeof((arr)[0]))
```

Comment:
Glad this file exists :) A more descriptive name like helpers or macros might be good.